parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/SerializationUtil.java

```diff
@@ -53,22 +53,13 @@ private SerializationUtil() { }
    * @throws IOException if there is an error while writing
    */
   public static void writeObjectToConfAsBase64(String key, Object obj, Configuration conf) throws IOException {
-    ByteArrayOutputStream baos = null;
-    GZIPOutputStream gos = null;
-    ObjectOutputStream oos = null;
-
-    try {
-      baos = new ByteArrayOutputStream();
-      gos = new GZIPOutputStream(baos);
-      oos = new ObjectOutputStream(gos);
-      oos.writeObject(obj);
-    } finally {
-      Closeables.close(oos);
-      Closeables.close(gos);
-      Closeables.close(baos);
+    try(ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+      try(GZIPOutputStream gos = new GZIPOutputStream(baos);
+            ObjectOutputStream oos = new ObjectOutputStream(gos)) {
+        oos.writeObject(obj);
+      }
+      conf.set(key, new String(Base64.encodeBase64(baos.toByteArray()), StandardCharsets.UTF_8));
```

Comment:
This code is not equivalent to the original one: `baos.toByteArray()` was invoked after closing the stream, now it is the other way around.

Reply:
This is true, but with the try-with-resource, we don't have a reference after closing the stream. Please note that the `.close()` on the `ByteArrayOutputStream` doesn't have any effect: https://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayOutputStream.html#close()
The important thing here is that the `ObjectOutputStream` and `GZIPOutputStream` are flushed (and closed).

Reply:
You're right, thanks for explaining.